ci/release-info-v2.py

```diff
@@ -0,0 +1,96 @@
+import json
+import os
+import re
+import sys
+from datetime import date
+from subprocess import Popen, PIPE
+
+def read_version(crate):
+    # Reads package version from workspace manifest file
+    channel = Popen(
+        ['cargo', 'read-manifest', '--manifest-path', './{}/Cargo.toml'.format(crate)],
+        stdout=PIPE
+    )
+
+    # parse json data
+    data = json.load(channel.stdout)
+    version = data['version']
+    return version
+
+def check_version(ver):
+    # Checks package version for matching with the current tag reference
+    if ref is not None and ref != 'refs/tags/v' + str(ver):
+        return 0
+    else:
+        return 1
+
+def print_error(crate, ver, v):
+    # print(crate, ver, v)
+    if not v:
+        print(
+            '::error file={path}::version {0} does not match release tag {1}'
+            .format(str(ver), ref, path = str(crate) + '/Cargo.toml')
+        )
+
+def bundle_version(crates):
+    # Checks package versions of the crates bundle for consistency with the given tag reference
+    bundle = list(map(read_version, crates))
+    consistency = list(map(check_version, bundle))
+    # print(crates, bundle, consistency)
+
+    # print errors for packages which versions didn't match tag reference
+    if not all(consistency):
+        list(map(print_error, crates, bundle, consistency))
+        sys.exit(1)
+    else:
+        version = bundle[0]
+        return version
```

Comment:
This block is not needed. Could be outside the else.

Reply:
This return actually used below for version variable used in the script main body.

Reply:
Yeah, I mean:
```python
if ____:
    return
return bundle[0]
```

Reply:
Yeah, thanks to mention that! I will add the check of consistency here, as I did already in futher version 3 of this script. Mainly bundle_version() return is conditioned by consistency check for versions of packages in bundle, versions must be similar, and only then the return of one consolidated version must be performed.